source/common/access_log/access_log_formatter.cc

```diff
@@ -301,6 +301,16 @@ RequestInfoFormatter::RequestInfoFormatter(const std::string& field_name) {
       return RequestInfo::Utility::formatDownstreamAddressNoPort(
           *request_info.downstreamRemoteAddress());
     };
+  } else if (field_name == "REQUESTED_SERVER_NAME") {
+    field_extractor_ = [](const RequestInfo::RequestInfo& request_info) {
+      absl::string_view requested_server_name;
+      if (!request_info.requestedServerName().empty()) {
+        return request_info.requestedServerName().data();
+      } else {
+        requested_server_name = UnspecifiedValueString;
+        return requested_server_name.data();
```

Comment:
The return type for field_extractor_ is std::string, so it seems to me this converts UnspecifiedValueString into a string_view and then back to std::string. Does `return UnspecifiedValueString` not work?

Reply:
I think if you make the lambda's return type explicit (`[](const RequestInfo::RequestInfo&) -> std::string { ... }` ) it'll probably work, but you might have to convert the absl::string_view in the other return explicitly as well.